internal/bufferedwrites/buffered_write_handler.go

```diff
@@ -134,3 +134,7 @@ func (wh *BufferedWriteHandler) WriteFileInfo() WriteFileInfo {
 		Mtime:     wh.mtime,
 	}
 }
+
+func (wh *BufferedWriteHandler) SignalUploadFailure() chan error {
+	return wh.uploadHandler.signalUploadFailure
```

Comment:
Not sure if this is good practice. UploadHandler private variables being exposed by another struct to other structs.

Reply:
moved to upload handler